java/IBAutomater/src/ibautomater/WindowEventListener.java

```diff
@@ -385,5 +395,61 @@ private boolean HandleFinancialAdvisorWarningWindow(Window window, int eventId)
 
         return false;
     }
+
+    private boolean HandleExitSessionSettingWindow(Window window, int eventId) throws Exception {
+        if (eventId != WindowEvent.WINDOW_ACTIVATED) {
+            return false;
+        }
+
+        String title = Common.getTitle(window);
+
+        if (title != null && title.contains("Exit Session Setting")) {
+            LocalDateTime time = java.time.LocalDateTime.now();
+
+            // disable logout/restart by setting logoff/restart time to the next day
+            time = time.minusMinutes(10);
+
+            // set new time
+            String timeText = DateTimeFormatter.ofPattern("hh:mm").format(time);
+            JTextField timeTextField = Common.getTextField(window, 0);
+
+            if (timeTextField == null) {
+                throw new Exception("Time text field not found");
+            }
+
+            this.automater.logMessage("Set time: [" + timeText + "]");
+            timeTextField.setText(timeText);
+
+            // set AM/PM
+            String formattedTime = DateTimeFormatter.ofPattern("hh:mm a").format(time);
+            String buttonText = formattedTime.endsWith("AM") ? "AM" : "PM";
+            JRadioButton radioButton = Common.getRadioButton(window, buttonText);
+
+            if (radioButton != null) {
+                this.automater.logMessage("Click radio button: [" + buttonText + "]");
+                radioButton.doClick();
+            }
+
+            buttonText = "Apply";
+            JButton button = Common.getButton(window, buttonText);
+
+            if (button != null) {
+                this.automater.logMessage("Click button: [" + buttonText + "]");
+                button.doClick();
+            }
+
+            buttonText = "OK";
+            button = Common.getButton(window, buttonText);
+
+            if (button != null) {
+                this.automater.logMessage("Click button: [" + buttonText + "]");
+                button.doClick();
+            }
```

Comment:
Should these `if (button != null)` have an else case which throws an exception?

Reply:
All exceptions thrown from here would only be logged at line 99 but no harm in adding an extra log line, will fix